lib/thread.c

```diff
@@ -498,6 +498,43 @@ DEFUN (clear_thread_cpu,
 	return CMD_SUCCESS;
 }
 
+static void show_thread_timers_helper(struct vty *vty, struct thread_master *m)
+{
+	const char *name = m->name ? m->name : "main";
+	char underline[strlen(name) + 1];
+	struct thread *thread;
+
+	memset(underline, '-', sizeof(underline));
+	underline[sizeof(underline) - 1] = '\0';
+
+	vty_out(vty, "\nShowing timers for %s\n", name);
+	vty_out(vty, "-------------------%s\n", underline);
+
+	frr_each (thread_timer_list, &m->timer, thread) {
+		vty_out(vty, "  %-50s%pTVMu\n", thread->hist->funcname,
```

Comment:
Nit: `"%pTVMu", &thread->u.sands` is the same as `"%pTH", thread` :smile:

(The full infodump for threads is `%pTHD`)

Reply:
When I used %pTH I got a whole bunch of other crap I didn't want to see.  Let me try it again and verify

Reply:
Hm, did I break something there? The long form output should be `%pTHD`, but I make mistakes too…

Reply:
```eva# show thread timers
Thread timers for zebra:

Showing timers for default
--------------------------
  rtadv_timer                                       {(thread *)0x5c89cf0 arg=0x5dbf210 timer  r=0.688      rtadv_timer() &rtadv->ra_timer from zebra/rtadv.c:2752}u
  zebra_route_map_update_timer                      {(thread *)0x62a0b40 arg=0x0 timer  r=1.175      zebra_route_map_update_timer() &zebra_t_rmap_update from zebra/zebra_routemap.c:1795}u
  if_zebra_speed_update                             {(thread *)0x5df6500 arg=0x5df5df0 timer  r=8.061      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df6e50 arg=0x5df67c0 timer  r=8.062      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df7720 arg=0x5df7090 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df80b0 arg=0x5df7a20 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df8a40 arg=0x5df82f0 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  zebra_evpn_mh_startup_delay_exp_cb                {(thread *)0x5cbaf00 arg=0x0 timer  r=172.907    zebra_evpn_mh_startup_delay_exp_cb() &(zrouter.mh_info)->startup_delay_timer from zebra/zebra_evpn_mh.c:3883}u
  if_zebra_speed_update                             {(thread *)0x5df51e0 arg=0x5df4af0 timer  r=8.041      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df9310 arg=0x5df8c80 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5df9be0 arg=0x5df9550 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5dfa5f0 arg=0x5df9f60 timer  r=8.067      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5dfaec0 arg=0x5dfa830 timer  r=8.068      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0x5dfb790 arg=0x5dfb100 timer  r=8.068      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  if_zebra_speed_update                             {(thread *)0xb73e4b0 arg=0xd15c280 timer  r=9.847      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u
  rib_sweep_route                                   {(thread *)0x628f040 arg=0x0 timer  r=113.767    rib_sweep_route() &zrouter.sweeper from zebra/main.c:440}u
  if_zebra_speed_update                             {(thread *)0x5df5b50 arg=0x5df54c0 timer  r=8.052      if_zebra_speed_update() &zebra_if->speed_update from zebra/interface.c:216}u

```

Reply:
(we got it worked out on slack, but in case someone ends up here in a year looking at something else: `%pTH` prints the timer only, `%pTHD` is the long debug output shown above, the `D` is for "Debug")

Reply:
ah I was using `%pTHD` I'll switch it over to `%pTH` as that I get the expected output